docs/dependent-apps.md

```diff
@@ -0,0 +1,289 @@
+# Using the Resin Supervisor for managing dependent applications
+
+Since version 2.5.0 the Resin Supervisor can act as a proxy for dependent apps.
+
+Only Supervisors after version 2.5.0 have this functionality, and some of the endpoints appeared in later versions (we've noted it down where this is the case).
+
+## What is a dependent application
+
+A **dependent application** is a resin application that targets devices not capable of interacting directly with the Resin API - the reasons can be several, the most common are:
+
+- no direct Internet capabilities
+- not able to run Resin OS (being a microcontroller, for example)
+
+The **dependent application** is scoped under a resin application, which gets the definition of **gateway application**.
+
+The **gateway application** is responsible for detecting, provisioning and managing **dependent devices** belonging to one of its **dependent applications**. This is possible leveraging a new set of endpoints exposed by the Resin Supervisor.
+
+A dependent application follows the same development cycle of a conventional resin application:
+
+- it binds to your git workspace via the **resin remote**
+- it consists in a Docker application
+- it offers the same environment and configuration variables management
+
+There are some differences:
+
+- it does not support Dockerfile templating
+- the Dockerfile must target a `x86` base image
+- the actual firmware/business logic must be stored in the `/assets` folder within the built docker image.
+
+  - You can either just `COPY` a pre-built artifact in that folder, or build your artifact at push time and then store it in the `/assets` folder.
+
+## How a dependent application works
+
+### Design
+
+Resin Supervisor exposes a REST interface when the data-flow goes from the gateway application to the Resin API (from now on, called **pull** dataflow) and a **Hook** mechanism when the data-flow goes from the Resin API to the gateway application (from now on, called **push** dataflow). The term **Hook** is meant as a convention of HTTP requests made from the Resin Supervisor to the gateway application
+
+
+### Hook
+
+It's worth noting the **Hook** APIs described in this document consist in the opposite direction flow: It's the Resin Supervisor consuming a REST interface exposed by the gateway application. These methods are described as **what the Resin Supervisor expects to find on a push dataflow**
+
+### Hook configuration
+
+You can point the supervisor where to find the hook server via a configuration variable.
+
+- `RESIN_DEPENDENT_DEVICES_HOOK_ADDRESS` _(defaults to `http://0.0.0.0:1337/v1/devices/`)_
```

Comment:
Might be good to point out here that the supervisor will append the dependent device uuid to every hook request URL.